### PR TITLE
Altered settings for slow machine based ARMv7

### PR DESCRIPTION
**Pull request by baytuch**
This fix for slow machine based ARMv7 SoC like BCM2835. Testet on the Raspberry PI 3 (Raspbian Stretch). The system allocate too much RAM. It also leads to a panic kernel, through overflow of VRAM.

- fix set new default value for client_mapblock_limit as 250

Debug:
max_loaded_blocks: 250
block_count_all: 257
max_loaded_blocks: 250
block_count_all: 258
max_loaded_blocks: 250
block_count_all: 384
max_loaded_blocks: 250
block_count_all: 409

### CODE REVIEW

File: src/defaultsettings.cpp [hunk 1]
```diff
@@ -508,5 +508,10 @@ void set_default_settings()
 		settings->setDefault("mono_font_size", "14");
 	}
 	// Tablets >= 6.0 use non-Android defaults for these settings
+#else
+	// Altered settings for slow machine based ARMv7 SoC like BCM2835
+#if defined(__ARM_32BIT_STATE)
+	settings->setDefault("client_mapblock_limit", "250");
+#endif
```
[the remark this answers is not in the document]
Comment (baytuch) [reply src/defaultsettings.cpp:515]: This limit is valid for operational systems compiled for Armv7, as a debian - for slow computers. Android is simply not able to work in such a configuration. I use this fix for build for my Raspberry PI 3.

**Review by SmallJoker — APPROVED** [review]
A dynamic solution based on total RAM would be more sophisticated solution, but considering generally low RAM on 32-bit ARM systems, I think this is acceptable.

**Comment by sfan5** [comment]
Unfortunately `__ARM_32BIT_STATE` does not detect what the SoC is *capable* of but rather what the OS is running at.
For example until recently Raspberry OS and all applications would be 32-bit even on the newest RPis with 8GB of RAM.

By the way Irrlicht has a [method](https://github.com/minetest/irrlicht/blob/df908ef4eaa7fd3292faa34d690edce40106da3c/include/IOSOperator.h#L35-L39) to retrieve the amount of system RAM you could use. Unfortunately it's broken because it uses a 32-bit number type...

**Comment by thesamesam** [comment]
> Unfortunately `__ARM_32BIT_STATE` does not detect what the SoC is _capable_ of but rather what the OS is running at. For example until recently Raspberry OS and all applications would be 32-bit even on the newest RPis with 8GB of RAM.
> 

Note that you wouldn't be able to use 8GB of RAM in Minetest in such a configuration. I think the idea here is to use ARM32 as a cheap proxy for "should we run in a resource-constrained mode?", but maybe it's better to have a configuration flag for it instead (and maybe suggest to use it on startup w/ ARM32 and some other situations)?

**Comment by sfan5** [comment]
Marking this as possible close because I dislike the approach, see the alternative I suggested.
No idea how @SmallJoker feels about this.

**Comment by SmallJoker** [comment]
I think the alternative is unusable due to unpredictable over-/underflows. To provide a better user experience, it would be nice to scale `client_mapblock_limit` and the view range on the system's capabilities. This PR would be simple yet very incomplete implementation.
It would be unfortunate to close this PR, but I agree that a generic solution is preferable.

**Comment by baytuch** [comment]
Automatic set limit - maybe, is good, but a have problem with allocating mem for GPU too (kernel panic for GPU driver). I think, really need some mechanism for this.

But my solution work good for PI 3 with limiting recourse, else - I have crash system and need hard reset of the device.

**Pull request closed without merging**